lib/id3c/cli/command/etl/manifest.py

```diff
@@ -126,6 +126,13 @@ def etl_manifest(*, db: DatabaseSession):
                 or collection_identifier.set_name in expected_identifier_sets["collections"], \
                     f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}" # type: ignore
 
+            # Validate sample collection date if available
+            collection_date = manifest_record.document.get("date")
+            if collection_date:
+                # These are the current date formats found in `receiving.manifest`
+                expected_date_formats = ("%m/%d/%Y", "%m/%d/%y", "%Y-%m-%d %H:%M:%S")
+                collection_date = validate_date(collection_date, expected_date_formats)
```

Comment:
I'm not sure it's sustainable to try to validate all datetime formats we might see. We'll have to update the code whenever a new format is used. Hardcoding formats also doesn't align with the broader vision of ID3C as not just for SFS. I'd be tempted to rely on Pg's extensive native date parsing instead. For example, the statements in `upsert_sample()` can use `date_or_null(%(collection_date)s)` when setting `collection_date`.

One downside would be losing the warnings about bogus dates, but I think that might actually be an upside in terms of ETL job noise. I don't think we want to be in the business of policing all those dates as they roll in. 🌊 We would still be able to do bulk QC by writing a query to find samples without a `collection_date` and compare it to the original manifest record's (unparsed) `date` (either from `sample.details` or `receiving.manifest`).

Reply:
I am 100% onboard with less ETL job noise. 😄 

I'll update `upsert_sample` to use our custom `date_or_null` function to set the collection date value.